src/test/compile-fail/not-panic-safe-2.rs

```diff
@@ -0,0 +1,22 @@
+// Copyright 2015 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+#![allow(dead_code)]
+#![feature(recover)]
+
+use std::panic::PanicSafe;
+use std::rc::Rc;
+
+fn assert<T: PanicSafe + ?Sized>() {}
+
+fn main() {
+    assert::<Rc<i32>>(); //~ ERROR: is not implemented
```

Comment:
Ok that didn't work (using `Shared`), because `Rc<T>` stores `Shared<RcBox<T>>` and `RcBox<T>` stores `UnsafeCell<usize>`, violating the "no unsafe cell" inside.

That means that to get these two pointers past the chcks we'll have to provide specific impls (which I was hoping to avoid doing), but seems fine to do for now.